src/App.tsx

```diff
@@ -8,7 +7,53 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+import { Status } from './types/types';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [query, setQuery] = useState('');
+  const [statusSelect, setStatusSelect] = useState('all');
+
+  const onTodoSelection = (todoId: number) => {
+    setSelectedTodo(todos.find(todo => todo.id === todoId) || null);
+  };
+
+  const handleModalClose = () => {
+    setSelectedTodo(null);
+  };
+
+  const searchByInput = (todoTitle: string, searchInput: string) => {
+    return todoTitle.toLowerCase().includes(searchInput.toLowerCase());
+  };
+
+  useEffect(() => {
+    getTodos().then(todo => {
+      setTodos(todo);
+      setIsLoading(false);
```

Comment:
Move it to finally so if there's an error in promise loading will be finished anyway.